py-polars/polars/dependencies.py

```diff
@@ -163,25 +165,26 @@ def _lazy_import(module_name: str) -> tuple[ModuleType, bool]:
     )
 
 
-def _NUMPY_TYPE(obj: Any) -> bool:
-    return _NUMPY_AVAILABLE and any(
-        "numpy." in str(o)
-        for o in (obj if inspect.isclass(obj) else obj.__class__).mro()
-    )
+@cache
+def _might_be(cls: type, type_: str) -> bool:
+    # infer whether the given class "might" be associated with the given
+    # module (in which case it's reasonable to do a real isinstance check)
+    try:
```

Comment:
Can you use https://github.com/pola-rs/polars/commit/4bd67da070e667b3cb2ee51cff9afa69c825d106 ?
Latest ruff, will flag it.